pkg/interceptors/bitbucket/bitbucket.go

```diff
@@ -26,20 +26,27 @@ import (
 	"google.golang.org/grpc/codes"
 )
 
-var _ triggersv1.InterceptorInterface = (*Interceptor)(nil)
+var _ triggersv1.InterceptorInterface = (*BaseInterceptor)(nil)
 
-type Interceptor struct {
+type BaseInterceptor struct {
 	SecretGetter interceptors.SecretGetter
 }
 
-func NewInterceptor(sg interceptors.SecretGetter) *Interceptor {
-	return &Interceptor{
+func NewInterceptor(sg interceptors.SecretGetter) *BaseInterceptor {
+	return &BaseInterceptor{
 		SecretGetter: sg,
 	}
 }
 
-func (w *Interceptor) Process(ctx context.Context, r *triggersv1.InterceptorRequest) *triggersv1.InterceptorResponse {
-	p := triggersv1.BitbucketInterceptor{}
+// Interceptor provides a webhook to intercept and pre-process events
+type Interceptor struct {
```

Comment:
This Interceptor struct defines the structure of the `[InterceptorParams](https://github.com/tektoncd/triggers/blob/d54faeb18914b42f0f8a02ae69a3287879194e5c/pkg/apis/triggers/v1beta1/interceptor_types.go#L36)` for this particular InterceptorImplementation....does this struct have to be called Interceptor? Maybe `InterceptorParams` or just `BitBucketParams` since this is specific to this interceptor?

Reply:
sure - this is subjective, I found it a bit confusing what the Interceptor struct meant here...really all it is is the set of parameters that this interceptor accepts. So, to me InterceptorParams or (event just `Params`) makes it a bit more descriptive

Reply:
I don't think feel too strongly about this, so feel free to keep the `Interceptor` name though in that case, please add a comment describing what the struct contents.

Reply:
@dibyom made the required changes